.github/workflows/pir_end_to_end_tests.yml

```diff
@@ -0,0 +1,163 @@
+name: PIR E2E Tests
+
+on:
+  workflow_dispatch:
+  schedule:
+     - cron: '0 3 * * 1-5' # 3AM UTC offsetted to legacy to avoid action-junit-report@v4 bug
+  pull_request:
+      
+jobs:
+  pir-e2e-tests:
+    name: PIR e2e tests
+    runs-on: ${{ matrix.runner }}
```

Comment:
Are you doing this because we plan to run on multiple versions later?

Reply:
It may make sense to test across different platforms, but it's probably a good idea to make sure tests run reliably on the current version before adding more.